Kitty stephan analysis

kittyzeng opened the pull request:
Updated our analysis function to include an edge case and wrote two test cases for it under TestAnalysisMethods.

mscuthbert commented:
Failing linting and has conflicts with main branch.  Please merge master to your own branch and fix conflicts and lint problems and request review again.  Great start!

kittyzeng commented:
We just updated our code:
- made function name more descriptive
- updated docstring to be more detailed
- removed unnecessary print statements
- resolved merge conflicts
- updated test cases
